modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/error/ConstructionError.scala

```diff
@@ -11,13 +11,23 @@ sealed trait ConstructionError extends NoStackTrace
 object ConstructionError {
   implicit class ConstructionErrorOps(ce: ConstructionError) {
     def toRosettaError: RosettaError = ce match {
-      case InvalidPublicKey     => RosettaError.InvalidPublicKey
-      case MalformedTransaction => RosettaError.MalformedTransaction
-      case UnsupportedOperation => RosettaError.UnsupportedOperation
+      case IncorrectNumberOfOperations => RosettaError.InvalidRequestBody("Exactly 2 Operations required")
+      case InvalidPublicKey            => RosettaError.InvalidPublicKey
+      case InvalidSuggestedFee         => RosettaError.InvalidRequestBody("Suggested Fee must be positive")
+      case MalformedTransaction        => RosettaError.MalformedTransaction
+      case MismatchedOperations        => RosettaError.InvalidRequestBody("Operations have mismatched Amount values")
+      case MissingAccountIdentifier    => RosettaError.InvalidRequestBody("Missing AccountIdentifier")
+      case SerializationError(reason)  => RosettaError.InternalError(s"Serialization failed: $reason")
+      case UnsupportedOperation        => RosettaError.UnsupportedOperation
     }
   }
 }
 
+case object IncorrectNumberOfOperations extends ConstructionError
```

Comment:
as above, if you want to specify the number of operations that are required, imo you should use a `case class IncorrectNumberOfOperations(required: Int) ...`